File: 250219/K개 중에 1개를 N번 뽑기/n-permutations-of-k-with-repetition.java
import java.util.*;

public class Main {
    static ArrayList<Integer> list = new ArrayList<>();
    static int k, n;
    
    static void choose(int digit){
        if(digit == n+1) {
            for(int i = 0; i < list.size(); i++){
                System.out.print(list.get(i) + " ");
            }
            System.out.println();
            return;
        }

        for(int select = 1; select <= k; select++){
            list.add(select);
            choose(digit+1);
            list.remove(list.size() - 1);
        }
    }

    public static void main(String[] args) {
        // Please write your code here.
        Scanner sc = new Scanner(System.in);
        k = sc.nextInt();
        n = sc.nextInt();

        choose(1);
    }
}

File: 260325/K개 중에 1개를 N번 뽑기/n-permutations-of-k-with-repetition.py
K, N = map(int, input().split())

# Please write your code here.
answer = []

def choose(curr_n):
    if (curr_n > N):
        print(*answer)
        return
    
    for i in range(1, K + 1):
        answer.append(i)
        choose(curr_n+1)
        answer.pop()

choose(1)